esptool/cli_util.py

```diff
@@ -294,7 +294,7 @@ def __init__(self, *args, **kwargs):
         self._eat_all_parser = None
         # Set the metavar dynamically based on the type's metavar
         if self.type and hasattr(self.type, "name"):
-            self.metavar = f"[{self.type.get_metavar(None) or self.type.name.upper()}]"
+            self.metavar = f"[{self.type.get_metavar(None, None) or self.type.name.upper()}]"
```

Comment:
This fix makes it work with click >= 8.2.0, but it does not work with older versions (see the failed tests). We would like to keep the compatibility. We should find a solution that works for both versions, specifically <8.2.0 and >=8.2.0. You can take inspiration here: https://github.com/sirosen/click/blob/55189e3ccf6677f3f350d4a1180c7c95142d79d7/docs/support-multiple-versions.md